### PR TITLE
Perceiver Transformer for Classification of ADNI Dataset - Rohan Kollambalath 46963765 

### PR DESCRIPTION
**Pull request by rkoll55**
# Pull Request for Perceiver Transformer on ADNI Dataset

## Overview
In this project I attempted to produce a perceiver transformer that classifies if MRI brain scans from the ADNI dataset have Alzheimer's disease or not. 

In the end after a lot of attempts of hyper tuning I achieved 57% accuracy on the test set which I attribute to various factors with the quality of data. 

I may attempt to change the data split and the way I encode images in the future. Please refer to my README for more details on implementation and explanation of results

## Folder Structure
- plots/
- recognition/
    - alzheimers_transformer_46963765/
        - dataset.py
        - modules.py
        - predict.py
        - train.py
    - README.md  

**plots**: Folder that contains figures including loss and accuracy plots  
**README.md**: Description of project, process methods and results  
**dataset.py**: Reads from dataset to create data loaders and visualise data  
**modules.py**: Contains the implantation of the perceiver transformer
**predict.py**: Contains functions to test accuracy of the model  
**train.py**: Driver script that runs and trains the model. 

## Usage
Download the ADNI brain dataset into the *alzheimers_transformer_46963765 folder* and run train.py with your desired hyperparameters.

Cheers   
Rohan Kollambalath   
46963765

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required**

**Difficulty**: 
- Hard
- Task 6: . Classify Alzheimer’s disease
- Perceiver transformer

**Readme**
- Excellent

**Commit messages**
- Good

**Code**
- Good structure
- Adequate comments

**Functionality/Performance**
- 57% classification accuracy
- Did patient level split
- the accuracy plot of testing data is flat, there might be some bugs so the model is not learning properly

**General comments**
- Good report
- Good plots
- No changes required

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure not present, needs own folder with name for all content -2
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, fix directory structure to allow merge -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks can be obtained +2 by fixing directory structure to allow merge, currently changes repo README etc. which must be undone.

**Comment by rkoll55** [comment]
Hey, made the changes as requested and should be good to merge. Cheers.

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
